hal/src/rcc.rs

```diff
@@ -504,6 +516,54 @@ fn cpu1_systick(rcc: &pac::RCC, cfgr: &pac::rcc::cfgr::R, src: SystClkSource) ->
     }
 }
 
+pub fn pclk1(rcc: &pac::RCC, cfgr: &pac::rcc::cfgr::R) -> Ratio<u32> {
+    let div: u32 = ppre_div(cfgr.ppre1().bits()).into();
+    hclk1(rcc, cfgr) / div
+}
+
+/// Calculate the current PCLK1 frequency in hertz
+///
+/// Fractional frequencies will be rounded down.
+///
+/// # Example
+///
+/// ```no_run
+/// use stm32wl_hal::{pac, rcc::pclk1_hz};
+///
+/// let dp: pac::Peripherals = pac::Peripherals::take().unwrap();
+///
+/// // without any initialization pclk1 will be 4MHz
+/// assert_eq!(pclk1_hz(&dp.RCC), 4_000_000);
+/// ```
+pub fn pclk1_hz(rcc: &pac::RCC) -> u32 {
+    let cfgr: pac::rcc::cfgr::R = rcc.cfgr.read();
+    pclk1(rcc, &cfgr).to_integer()
+}
+
+pub fn pclk2(rcc: &pac::RCC, cfgr: &pac::rcc::cfgr::R) -> Ratio<u32> {
```

Comment:
To match the others for now:
```suggestion
fn pclk2(rcc: &pac::RCC, cfgr: &pac::rcc::cfgr::R) -> Ratio<u32> {
```